src/lib/crypto/kimchi_bindings/wasm/src/wasm_flat_vector.rs

```diff
@@ -1,3 +1,10 @@
+//! The flat vector is a vector of fixed-size elements that we want to expose directly to js-of-ocaml
+//! (so that we can access a `Vec<Field>` cheaply,
+//! by just passing a pointer to a continuous memory region instead of copying.
+//! The wasmvector is a normal heap-allocated vector,
+//! where we leave it on the rust heap and just keep a pointer around.
+//! We use flat for fields, normal for gates etc.
```

Comment:
Thanks for documenting this!! However, the following leaves me confused:
```
(so that we can access a `Vec<Field>` cheaply,
by just passing a pointer to a continuous memory region instead of copying.
```
apart from a missing parenthesis, to me this sounds like it refers to the wasmvector, not the flat vector. I thought the flat vector was the full byte content and not a pointer, while the wasmvector is a pointer?

Reply:
I believe I copied this verbatim from what @mrmr1993 told me :D